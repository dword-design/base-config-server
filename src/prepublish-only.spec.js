import execa from 'execa'
import withLocalTmpDir from 'with-local-tmp-dir'
import P from 'path'
import outputFiles from 'output-files'

export default {
  valid: () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'package.json': JSON.stringify(
          {
            baseConfig: require.resolve('.'),
          },
          undefined,
          2
        ),
        'src/index.js': 'export default 1 |> x => x * 2',
      })
      await execa.command('base prepare')
      await execa.command('base prepublishOnly')
      expect(require(P.resolve('dist'))).toEqual(2)
    }),
}
