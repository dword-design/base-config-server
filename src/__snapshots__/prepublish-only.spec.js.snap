// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepublish-only valid 1`] = `
"var _;
export default (_ = 1, _ * 2);"
`;
